src/utils/reducers/ShoppingCartReducer.js

```diff
@@ -0,0 +1,71 @@
+export const ShoppingStateEnums = {
+  DELETE_ITEMS: 0,
+  MINUS_ITEM: 1,
+  ADD_ITEM: 2,
+  PLUS_ITEM: 3,
+};
+
+export const initialStateShoppingCart = {
+  count: 0,
+  cartInfo: []
+};
+
+const shoppinCartReducer = (state, action) => {
+  const { type, payload } = action;
+  let copyCartInfo = [...state.cartInfo];
+  const indexItem = state.cartInfo.findIndex(
+    (cartItem) => cartItem.id == payload.id
+  );
+
+  switch (type) {
+    case ShoppingStateEnums.DELETE_ITEMS:
+      copyCartInfo = copyCartInfo.filter(
+        (cartItem) => cartItem.id != payload.id
+      );
+
+      return {
+        count: state.count - parseInt(payload.quantity),
+        cartInfo: copyCartInfo,
+      };
+
+    case ShoppingStateEnums.MINUS_ITEM:
+
+      if (copyCartInfo[indexItem].quantity > 1) {
+        copyCartInfo[indexItem].quantity -= 1;
+      } else {
+        copyCartInfo = copyCartInfo.filter(
+          (cartItem) => cartItem.id != payload.id
+        );
+      }
+
+      return {
+        count: state.count - 1,
+        cartInfo: copyCartInfo,
+      };
+
+    case ShoppingStateEnums.PLUS_ITEM:
+    console.log(" ShoppingStateEnums.PLUS_ITEM");
+    state.cartInfo[indexItem].quantity += 1;
```

Comment:
It is recommendable not update the state, instead create a const
```suggestion
   const newQty = state.cartInfo[indexItem].quantity + 1;
```

Reply:
Apply this comment for all the file

Reply:
done